source/solvers/heat_transfer.cc

```diff
@@ -217,38 +241,82 @@ HeatTransfer<dim>::assemble_system(
             {
               // Store JxW in local variable for faster access
               const double JxW = fe_values_ht.JxW(q);
+
+              const auto velocity = velocity_values[q];
+
+
+              // Calculation of the magnitude of the velocity for the
+              // stabilization parameter
+              const double u_mag = std::max(velocity.norm(), 1e-12);
+
+              // Calculation of the GLS stabilization parameter. The
+              // stabilization parameter used is different if the simulation is
+              // steady or unsteady. In the unsteady case it includes the value
+              // of the time-step
+              const double tau =
+                is_steady(time_stepping_method) ?
+                  1. / std::sqrt(std::pow(2. * rho_cp * u_mag / h, 2) +
+                                 9 * std::pow(4 * alpha / (h * h), 2)) :
+                  1. / std::sqrt(std::pow(sdt, 2) +
+                                 std::pow(2. * rho_cp * u_mag / h, 2) +
+                                 9 * std::pow(4 * alpha / (h * h), 2));
+
               // Gather the shape functions and their gradient
               for (unsigned int k : fe_values_ht.dof_indices())
                 {
                   phi_T[k]      = fe_values_ht.shape_value(k, q);
                   grad_phi_T[k] = fe_values_ht.shape_grad(k, q);
+                  hess_phi_T[k] = fe_values_ht.shape_hessian(k, q);
+
+                  laplacian_phi_T[k] = trace(hess_phi_T[k]);
                 }
 
+
+
               for (const unsigned int i : fe_values_ht.dof_indices())
                 {
                   const auto phi_T_i      = phi_T[i];
                   const auto grad_phi_T_i = grad_phi_T[i];
 
+
                   if (assemble_matrix)
                     {
                       for (const unsigned int j : fe_values_ht.dof_indices())
                         {
-                          const auto phi_T_j      = phi_T[j];
-                          const auto grad_phi_T_j = grad_phi_T[j];
+                          const auto phi_T_j           = phi_T[j];
+                          const auto grad_phi_T_j      = grad_phi_T[j];
+                          const auto laplacian_phi_T_j = laplacian_phi_T[j];
+
+
 
-                          // Weak form for : - k * laplacian T + rho * cp * u *
-                          //                      gradT - f - grad(u)*grad(u) =0
+                          // Weak form for : - k * laplacian T + rho * cp *
+                          // u
+                          // *
+                          //                      gradT - f -
+                          //                      grad(u)*grad(u) =0
```

Comment:
why is this equation on 5 lines rather than 2 as before?

Reply:
Error due to automatic indentation. i'll fix that.